src/lib.rs

```diff
@@ -176,3 +196,153 @@ fn no_lazy_static_rule() -> Rule {
         .build()
         .unwrap()
 }
+
+#[macro_export]
+macro_rules! assert_node_kind {
+    ($node:expr, $kind:literal $(,)?) => {{
+        assert_eq!($node.kind(), $kind);
+        $node
+    }};
+}
+
+fn get_constrained_type_parameter_name(node: Node) -> Node {
```

Comment:
So none of these "really" belong here (probably in the `tree_sitter_lint` crate at all)

They're just attempts to not horribly factor the code needed for the new rule's logic

It does seem like one would end up wanting to be able to maybe build up (mostly per-language-specific?) eg "tree-sitter AST node traversal" helpers

But that feels like it's more in the domain of the "consumer" of `tree_sitter_lint` than of `tree_sitter_lint` itself

Overall this code is reminding a fair amount of both the snippets of tree-sitter-based Lua code and of ESLint rule/plugin code. Which is good